classes/local/store/object_file_system.php

```diff
@@ -421,6 +429,13 @@ public function xsendfile_file(stored_file $file): bool {
 
             return $this->redirect_to_presigned_url($contenthash, headers_list());
         }
+
+        if ($this->externalclient->support_presigned_urls()) {
```

Comment:
as discussed move this logic into serve_range_request() which will be in each store

Reply:
Done